app/assets/javascripts/collections/story_collection.js

```diff
@@ -18,7 +18,49 @@ module.exports = Backbone.Collection.extend({
     this.labels = [];
   },
 
-  saveSorting: function(columnName) {
+  calculateNewPosition: function(previous_story_id, next_story_id) {
+    var newPosition;
+    if (!_.isUndefined(previous_story_id)) {
+      newPosition = this.calculatePositionAfter(previous_story_id);
```

Comment:
Instead of having this variable, what do you think about simply returning here? Then you wouldn't need the `elses` and would make it simpler like that:

```js
if (!_.isUndefined(previous_story_id)) {
   return this.calculatePositionAfter(previous_story_id);
} 

if (!_.isUndefined(next_story_id)) {
  return this.calculatePositionBefore(next_story_id);
}

if (this.length !== 1) {
  throw new Error("Unable to determine previous or next story id for dropped story");
}
```

Also, prefer not to `throw` a string, wrap it inside an `Error` instance 😄